File: backend/src/main/java/com/lifeos/telemetry/session/dto/SessionDto.java
package com.lifeos.telemetry.session.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.*;

import java.time.Instant;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SessionDto {

    @NotBlank(message = "source is required")
    private String source;

    @NotBlank(message = "type is required")
    private String type;

    @NotBlank(message = "packageName is required")
    private String packageName;

    @NotBlank(message = "title is required")
    private String title;

    private Instant startTime;

    private Instant endTime;

    private Long durationMillis;
}

File: android/app/src/main/java/com/karthik/lifeos/tracker/sync/dto/SessionDto.kt
package com.karthik.lifeos.tracker.sync.dto

import java.time.Instant

/**
 * DTO matching the backend's SessionDto.
 * Maps Android ActivitySession fields to the backend's expected shape.
 */
data class SessionDto(
    val source: String,
    val type: String,
    val packageName: String,
    val title: String,
    val startTime: String,     // ISO-8601 Instant string
    val endTime: String,       // ISO-8601 Instant string
    val durationMillis: Long
) {
    companion object {
        /**
         * Convert epoch millis to ISO-8601 Instant string for backend compatibility.
         * Backend uses java.time.Instant which Jackson deserializes from ISO strings.
         */
        fun fromEpochMillis(millis: Long): String {
            return Instant.ofEpochMilli(millis).toString()
        }
    }
}
